services/src/main/java/org/keycloak/protocol/oidc/grants/device/endpoints/DeviceEndpoint.java

```diff
@@ -286,12 +294,6 @@ private Response createVerificationPage(String errorMessage) {
     }
 
     private Response processVerification(OAuth2DeviceCodeModel deviceCode, String userCode) {
-        long expiresIn = deviceCode.getExpiration() - Time.currentTime();
-        if (expiresIn < 0) {
-            event.error(Errors.EXPIRED_OAUTH2_USER_CODE);
-            return createVerificationPage(Messages.OAUTH2_DEVICE_INVALID_USER_CODE);
-        }
-
```

Comment:
This was redundant with the `isExpired` check already happening in `verifyUserCode`. This condition would never evaluate to true. It's also less readable than the `isExpired()` method on `OAuth2DeviceCodeModel` which has the exact same logic.

An interesting aspect here is that it was duplicated and also subtly different–it used a different event code _and_ verification page message than the logic in `verifyUserCode` for this same condition. This diff keeps the same behavior as before but it's not exactly clear which message was intended.